slimt/Annotation.cc

```diff
@@ -74,4 +77,72 @@ void AnnotatedText::record_existing_sentence(
   annotation.token_begin_.push_back(text.size());
 }
 
+void AnnotatedText::to(Encoding encoding) {
+  if (encoding == Encoding::UTF8) {
```

Comment:
It's desirable to make this function _idempotent_. i.e, successive applications behave as one would expect.

```cpp
annotated.to(Encoding::UTF8); // Converts to UTF-8
annotated.to(Encoding::UTF8); // Successive same call, should not change, because in UTF-8.
annotated.to(Encoding::Byte); // Convert back to Byte
annotated.to(Encoding::Byte); // Successive same call, no change.
```

This requires storing `Encoding encoding_` as a property of annotation, and changed bookkeeping what encoding we're holding for annotation. 

The above also allows you to also write `else`. In general `if` without `else` is programmer-error. While this appears to be working for us now, we are creating broken paths when we don't do else that can bite us later.